apps/heft/src/pluginFramework/StaticFileSystemAdapter.ts

```diff
@@ -0,0 +1,234 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import type { ReaddirAsynchronousMethod, ReaddirSynchronousMethod } from '@nodelib/fs.scandir';
+import type { StatAsynchronousMethod, StatSynchronousMethod } from '@nodelib/fs.stat';
```

Comment:
Can't you just use `typeof fs.stat`, `typeof fs.statSync`, `typeof fs.readdir`, `typeof fs.readdirSync`?

Reply:
The type on fs is relatively complicated. Also, these are the types used by `FileSystemAdapter`, which we're extending in the first place. Since we're bringing them in as type imports and the packages are devdependencies, it felt reasonable to consume these in this manner.